sympy/physics/units/util.py

```diff
@@ -33,64 +33,79 @@ def _get_conversion_matrix_for_expr(expr, target_units, unit_system):
-
-    Conversion to Planck units:
-
-    >>> convert_to(atomic_mass_constant, [gravitational_constant, speed_of_light, hbar]).n()
-    7.62963085040767e-20*gravitational_constant**(-0.5)*hbar**0.5*speed_of_light**0.5
-
+def convert_to(expr, target_units, unit_system = "SI", checkUnits = True):
     """
+        Convert ``expr`` to the same expression with all of its units and quantities
+        represented as factors of ``target_units``, whenever the dimension is compatible.
+
+        ``target_units`` may be a single unit/quantity, or a collection of
+        units/quantities.
+
+        If ``checkUnits`` is set to true the function raises an exception if dimensions
+        of the converted expression are not compatable with the dimensions of input ``expr``.
+        Such an error is an indication the ``target_units`` are not compatible with the input ``expr``.
+
+        Examples
+        ========
+
+        >>> from sympy.physics.units import speed_of_light, meter, gram, second, day
+        >>> from sympy.physics.units import mile, newton, kilogram, atomic_mass_constant
+        >>> from sympy.physics.units import kilometer, centimeter
+        >>> from sympy.physics.units import gravitational_constant, hbar
+        >>> from sympy.physics.units import convert_to
+        >>> convert_to(mile, kilometer)
+        25146*kilometer/15625
+        >>> convert_to(mile, kilometer).n()
+        1.609344*kilometer
+        >>> convert_to(speed_of_light, meter/second)
+        299792458*meter/second
+        >>> convert_to(day, second)
+        86400*second
+        >>> 3*newton
+        3*newton
+        >>> convert_to(3*newton, kilogram*meter/second**2)
+        3*kilogram*meter/second**2
+        >>> convert_to(atomic_mass_constant, gram)
+        1.660539060e-24*gram
+
+        Conversion to multiple units:
+
+        >>> convert_to(speed_of_light, [meter, second])
+        299792458*meter/second
+        >>> convert_to(3*newton, [centimeter, gram, second])
+        300000*centimeter*gram/second**2
+
+        Conversion to Planck units:
+
+        >>> convert_to(atomic_mass_constant, [gravitational_constant, speed_of_light, hbar]).n()
+        7.62963085040767e-20*gravitational_constant**(-0.5)*hbar**0.5*speed_of_light**0.5
+
+        """
+    from sympy import Matrix
     from sympy.physics.units import UnitSystem
     unit_system = UnitSystem.get_unit_system(unit_system)
 
     if not isinstance(target_units, (Iterable, Tuple)):
         target_units = [target_units]
 
     if isinstance(expr, Add):
-        return Add.fromiter(convert_to(i, target_units, unit_system) for i in expr.args)
+        return Add.fromiter(convert_to(i, target_units, unit_system, checkUnits) for i in expr.args)
 
     expr = sympify(expr)
 
+    convertedExpr = _converter(expr, target_units, unit_system)
+
+    if checkUnits:
+        testMatrix = _get_conversion_matrix_for_expr(convertedExpr, [expr], unit_system)
+        if testMatrix != Matrix([[1]]):
```

Comment:
To me, this looks like fixing a bug hidden somewhere by adding a check.

This is actually OK as a temporary fix, but ideally it would be nice to find where the bug arises (i.e. the expression returned by the function `_converter` should be correct). I mean, this fix is simply hiding the bug...